spec/support/query_spy.cr

```diff
@@ -0,0 +1,12 @@
+# NOTE: This is only for testing if this is called during a query
+# TODO: Remove once a proper mocking shard is built
```

Comment:
Nice. I think we have a few other random "private" classes that do stuff like this that we could probably make a whole directory for this sort of stuff.